custom_model_runner/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py

```diff
@@ -131,12 +131,13 @@ def _materialize(self, parent_data_objs, user_data):
             transformed_df = transformed_output[0]
             transformed_df.to_csv(output_filename, index=False)
         else:
-            predictions = self._predictor.predict(
+            predictions, extra_df = self._predictor.predict(
```

Comment:
I'm putting this question here to make discussion easier, but it's a generic question

Why do we need to split and then reform the dataframe?  current new flow seems to be:

1. predict makes a df with sometimes extra columns
2. we split this into predictions and `other`
3. then in our response we combine `predictions` and `other` and then return that.

why not skip all of these steps and just return the original dataframe?

Reply:
This is how I started, but then figured out that there are many validations and data marshaling for the predictions themselves. There are assumptions for how the prediction DF should look like. I eventually found that the easiest way to go to is to split it at the bottom layer and recombined it where needed.
In any case, for the scoring server, the result should be serialized into a certain schema.

Reply:
please use a dataclass here and update type-hints to use it.

something like

```
@dataclass
class Predictions:
    prediction_dataframe: pd.DataFrame
    extra_dataframe: Optional[pd.DataFrame]
```

and this exposes some naming issues which expose some other questions

1. _what_ is getting returned by `predict`  it's Predictions and .... what?  what is the name of this combined thing?
2. what is `extra_dataframe`?  what is the point of it?  what does it do?  please update the name to reflect that